umbraco-cms/tutorials/creating-an-xml-site-map.md

```diff
@@ -208,146 +232,169 @@ If we create a helper called `RenderSiteMapUrlEntriesForChildren` that accepts a
+Finally, we need the helper to check the **Excluded Document Types** list on the XmlSiteMap Document Type.
 
-... we did but we haven't checked this value in the helper yet.
+1. Open the XmlSiteMap Template.
+2. Add the following code snippets within the first set of curly brackets:
 
-```csharp
-string excludedDocumentTypeList = Model.Value<string>("excludedDocumentTypes");
-string[] excludedDocumentTypes = (!String.IsNullOrEmpty(excludedDocumentTypeList)) ? excludedDocumentTypeList.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries).Select(p => p.Trim()).ToArray() : new string[] { };
-```
+    ```csharp
+    // Get the value from the excludedDocumentTypes property as a 'string'
+    string excludedDocumentTypeList = Model.Value<string>("excludedDocumentTypes");
+    // Separate the values into separate Document Types and add them to an 'array'
+    string[] excludedDocumentTypes = (!String.IsNullOrEmpty(excludedDocumentTypeList)) ? excludedDocumentTypeList.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries).Select(p => p.Trim()).ToArray() : new string[] { };
+    ```
 
-now we can pass this value into our helper
+3. Update the `RenderSiteMapUrlEntriesForChildren` helper as shown below to pass in the array:
 
-```csharp
-void RenderSiteMapUrlEntriesForChildren(IPublishedContent parentPage)
-{
-    foreach (var page in parentPage.Children.Where(f => !excludedDocumentTypes.Contains(f.ContentType.Alias) && !f.Value<bool>("hideFromXmlSiteMap")))
+    ```csharp
+    void RenderSiteMapUrlEntriesForChildren(IPublishedContent parentPage)
     {
-        @RenderSiteMapUrlEntry(page)
-        if (page.Level < maxSiteMapDepth && page.Children.Any(f => !excludedDocumentTypes.Contains(f.ContentType.Alias) && !f.Value<bool>("hideFromXmlSiteMap")))
+        // Filter the query based on the excludedDocumentTypes value
+        foreach (var page in parentPage.Children.Where(f => !excludedDocumentTypes.Contains(f.ContentType.Alias) && !f.Value<bool>("hideFromXmlSiteMap")))
         {
-            @RenderSiteMapUrlEntriesForChildren(page, maxSiteMapDepth, excludedDocumentTypes)
+            RenderSiteMapUrlEntry(page);
+            if (page.Level < maxSiteMapDepth && page.Children.Any(f => !f.Value<bool>("hideFromXmlSiteMap")))
+            {
+                RenderSiteMapUrlEntriesForChildren(page, maxSiteMapDepth, excludedDocumentTypes);
+            }
         }
     }
+    ```
 
-}
-```
+Visit the URL of your sitemap page (`http://yoursite.com/sitemap`) to render a complete sitemap for your site.
 
-### The finished sitemap template
+It contains an entry for each page that is
+
+* Not **hidden**,
+* Not based on an **excluded Document Type**, and
+* Located within the bounds of the defined **depth**.
+
+## The finished sitemap template
+
+**Add this to an expandable.**
 
 ```csharp
```

Comment:
This code block needs to be updated. It contains the old code.

Reply:
I am not entirely sure I understand what you mean about "the old code"?
Could you perhaps be a bit more specific as to where this code block need to be updated?

Reply:
I was confused with the last part of the code as it threw an error for me. Now that I run the last part of the code from the sample below it works fine 💪